### PR TITLE
edited spring xsd regex

### PR DESCRIPTION
**Pull request by bilalyasar**
closes https://github.com/hazelcast/hazelcast/issues/6534

### CODE REVIEW

**Comment by mesutcelik** [comment]
verify

**Comment by mesutcelik** [comment]
Do we have a test for this particular change? Can you please send a backport as well?

File: hazelcast-spring/src/main/resources/hazelcast-spring-3.6.xsd [hunk 1]
```diff
@@ -2007,7 +2007,7 @@
 
     <xs:simpleType name="parameterizedValueType">
         <xs:restriction base="xs:string">
-            <xs:pattern value="(\$|\#)\{([^=^:]+)\}"/>
+            <xs:pattern value="[$#]\{([^=^:]+)\}"/>
```
Comment (mesutcelik) [inline hazelcast-spring/src/main/resources/hazelcast-spring-3.6.xsd]: do we really need to exclude those chars `^=^:`? If yes, do we need to have some other?

Comment (bilalyasar) [inline hazelcast-spring/src/main/resources/hazelcast-spring-3.6.xsd]: i didn't remember real cause, but we can leave as it is.

**Pull request merged**